source/common/http/http1/codec_impl.cc

```diff
@@ -539,39 +559,54 @@ Http::Status ConnectionImpl::innerDispatch(Buffer::Instance& data) {
   return Http::okStatus();
 }
 
-size_t ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
+Envoy::StatusOr<size_t> ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
+  ASSERT(codec_status_.ok() && dispatching_);
   ssize_t rc = http_parser_execute(&parser_, &settings_, slice, len);
+  if (!codec_status_.ok()) {
+    return codec_status_;
+  }
   if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK && HTTP_PARSER_ERRNO(&parser_) != HPE_PAUSED) {
     sendProtocolError(Http1ResponseCodeDetails::get().HttpCodecError);
-    throw CodecProtocolException("http/1.1 protocol error: " +
-                                 std::string(http_errno_name(HTTP_PARSER_ERRNO(&parser_))));
+    // Avoid overwriting the codec_status_ set in the callbacks.
+    ASSERT(codec_status_.ok());
+    codec_status_ = codecProtocolError("http/1.1 protocol error: " +
```

Comment:
I realize this was pre-existing, but switch to using StrCat here?

Reply:
done